Add Dockerfiles and deployment guides for multiple services

farmanahmed888 opened the pull request.

farmanahmed888 committed:
Add Dockerfiles and deployment guides for multiple services

farmanahmed888 committed:
Refactor Dockerfiles to remove comments and standardize dependency installation using npm ci

Soumabha commented on ConferenceV2/docs/deployment.md:
```diff
@@ -0,0 +1,46 @@
+# ConferenceV2 Staging Deployment Guide
+
+## 1. Platform Choice
+- Recommended: [Render](https://render.com) or [Railway](https://railway.app)
```
Actually Railway has only a 30d trial plan so better to keep alternatives which are 'always' free.

Soumabha commented on ConferenceV2/docs/deployment.md:
```diff
@@ -0,0 +1,46 @@
+- Confirm `requirements.txt` lists all dependencies
+- Dockerfile is present in the project root
+
+## 3. Environment Variables
```
Could we keep the list of environment variables in the existing README for setup itself?

Soumabha commented on IVRv2/docs/deployment.md:
```diff
@@ -0,0 +1,65 @@
+- GitHub account and repository access
+- MongoDB Atlas staging connection string
+- Vonage and Azure credentials for staging
+- Free account on Render (https://render.com) or Railway (https://railway.app)
```
Actually Railway has only a 30d trial plan so better to keep alternatives which are 'always' free.

Soumabha commented on teacher-webapp/Dockerfile:
```diff
@@ -0,0 +1,13 @@
+
+COPY package*.json ./
+
+RUN npm ci --only=production
```
The following error occurs and requires syncing of the package.json and package-lock.json. Please fix.
```
1.590 npm error A complete log of this run can be found in: /root/.npm/_logs/2025-09-16T07_15_48_352Z-debug-0.log
------
Dockerfile:7
--------------------
   5 |     COPY package*.json ./
   6 |
   7 | >>> RUN npm ci --only=production
   8 |
   9 |     COPY . .
--------------------
ERROR: failed to build: failed to solve: process "/bin/sh -c npm ci --only=production" did not complete successfully: exit code: 1
```

Soumabha commented on websocket-service/docs/deployment.md:
```diff
@@ -0,0 +1,50 @@
+# Staging Deployment Guide: websocket-service
+
+## Platform Recommendation
+Use [Render](https://render.com/) or [Railway](https://railway.app/) for free Node.js deployments. Both support Docker and environment variables.
```
Railway gives usage only till 30d as per the trial plan. We can probably suggest some other alternatives as more permanent choices - Vercel is quite good.

Soumabha commented on websocket-service/docs/deployment.md:
```diff
@@ -0,0 +1,50 @@
+- Azure Blob Storage credentials
+
+## Environment Variables
+Set these in the platform dashboard (do not commit secrets):
```
Lets reference the environment variables to be in README.md for the repository; Duplication of the variables will make the updates more cumbersome and error prone.

Soumabha commented on ContentWebApp/Dockerfile:
```diff
@@ -0,0 +1,15 @@
+
+COPY package*.json ./
+
+RUN npm ci --only=production
```
I face the following error when building ContentWebApp - mostly is something to do with the synchronization of package.json and package-lock.json

```
 => ERROR [4/5] RUN npm ci --only=production                                                                                                                                                                                                                              1.9s
------
 > [4/5] RUN npm ci --only=production:
0.315 npm warn config only Use `--omit=dev` to omit dev dependencies from the install.
1.906 npm warn EBADENGINE Unsupported engine {
1.906 npm warn EBADENGINE   package: '@azure/core-auth@1.10.0',
1.906 npm warn EBADENGINE   required: { node: '>=20.0.0' },
1.906 npm warn EBADENGINE   current: { node: 'v18.20.8', npm: '10.8.2' }
1.906 npm warn EBADENGINE }
1.906 npm warn EBADENGINE Unsupported engine {
1.906 npm warn EBADENGINE   package: '@azure/core-client@1.10.0',
1.906 npm warn EBADENGINE   required: { node: '>=20.0.0' },
1.906 npm warn EBADENGINE   current: { node: 'v18.20.8', npm: '10.8.2' }
1.906 npm warn EBADENGINE }
1.906 npm warn EBADENGINE Unsupported engine {
1.906 npm warn EBADENGINE   package: '@azure/core-rest-pipeline@1.22.0',
1.906 npm warn EBADENGINE   required: { node: '>=20.0.0' },
1.906 npm warn EBADENGINE   current: { node: 'v18.20.8', npm: '10.8.2' }
1.906 npm warn EBADENGINE }
1.907 npm warn EBADENGINE Unsupported engine {
1.907 npm warn EBADENGINE   package: '@azure/core-tracing@1.3.0',
1.907 npm warn EBADENGINE   required: { node: '>=20.0.0' },
1.907 npm warn EBADENGINE   current: { node: 'v18.20.8', npm: '10.8.2' }
1.907 npm warn EBADENGINE }
1.907 npm warn EBADENGINE Unsupported engine {
1.907 npm warn EBADENGINE   package: '@azure/core-util@1.13.0',
1.907 npm warn EBADENGINE   required: { node: '>=20.0.0' },
1.907 npm warn EBADENGINE   current: { node: 'v18.20.8', npm: '10.8.2' }
1.907 npm warn EBADENGINE }
1.907 npm warn EBADENGINE Unsupported engine {
1.907 npm warn EBADENGINE   package: '@azure/core-xml@1.5.0',
1.907 npm warn EBADENGINE   required: { node: '>=20.0.0' },
1.907 npm warn EBADENGINE   current: { node: 'v18.20.8', npm: '10.8.2' }
1.907 npm warn EBADENGINE }
1.907 npm warn EBADENGINE Unsupported engine {
1.907 npm warn EBADENGINE   package: '@azure/logger@1.3.0',
1.907 npm warn EBADENGINE   required: { node: '>=20.0.0' },
1.907 npm warn EBADENGINE   current: { node: 'v18.20.8', npm: '10.8.2' }
1.907 npm warn EBADENGINE }
1.908 npm warn EBADENGINE Unsupported engine {
1.908 npm warn EBADENGINE   package: '@azure/storage-blob@12.28.0',
1.908 npm warn EBADENGINE   required: { node: '>=20.0.0' },
1.908 npm warn EBADENGINE   current: { node: 'v18.20.8', npm: '10.8.2' }
1.908 npm warn EBADENGINE }
1.909 npm warn EBADENGINE Unsupported engine {
1.909 npm warn EBADENGINE   package: '@azure/storage-common@12.0.0',
1.909 npm warn EBADENGINE   required: { node: '>=20.0.0' },
1.909 npm warn EBADENGINE   current: { node: 'v18.20.8', npm: '10.8.2' }
1.909 npm warn EBADENGINE }
1.911 npm warn EBADENGINE Unsupported engine {
1.911 npm warn EBADENGINE   package: '@typespec/ts-http-runtime@0.3.0',
1.911 npm warn EBADENGINE   required: { node: '>=20.0.0' },
1.911 npm warn EBADENGINE   current: { node: 'v18.20.8', npm: '10.8.2' }
1.911 npm warn EBADENGINE }
1.920 npm error code EUSAGE
1.920 npm error
1.920 npm error `npm ci` can only install packages when your package.json and package-lock.json or npm-shrinkwrap.json are in sync. Please update your lock file with `npm install` before continuing.
1.920 npm error
1.920 npm error Invalid: lock file's typescript@5.9.2 does not satisfy typescript@4.9.5
1.920 npm error
1.920 npm error Clean install a project
```

farmanahmed888 replied:
Please update your lock file with `npm install` before continuing.
this is from the logs above 
can you run 
`npm install`
and then try to build docker file

farmanahmed888 committed:
Refactor Dockerfiles for multiple services to use multi-stage builds and update deployment documentation for clarity and consistency

farmanahmed888 committed:
Refactor Dockerfiles to standardize dependency installation using npm ci --omit=dev and improve structure across multiple services

farmanahmed888 committed:
added package lock because render deployment is failing

farmanahmed888 committed:
Remove web-vitals from devDependencies and add it to dependencies in package.json

farmanahmed888 committed:
Remove fallback URL for API base URL in Login component

farmanahmed888 committed:
Refactor Dockerfiles to remove multi-stage build and simplify the structure for both ContentWebApp and teacher-webapp as it is a service not a static website

farmanahmed888 committed:
Revert "Refactor Dockerfiles to remove multi-stage build and simplify the structure for both ContentWebApp and teacher-webapp as it is a service not a static website"

This reverts commit f65576bd941977896982f309dac7b1fe374ae33c.

farmanahmed888 committed:
Update package.json to move json-server and buffer to dependencies

farmanahmed888 committed:
Refactor Dockerfile to simplify the build process by removing multi-stage build and using npm start for serving the application

farmanahmed888 committed:
Remove .env.example file and update Dockerfile to streamline build and serve process using multi-stage builds

farmanahmed888 committed:
Add render.yaml configuration for ContentWebApp with environment variables

farmanahmed888 committed:
Add render.yaml configuration for ContentWebApp with environment variables

farmanahmed888 committed:
Add rootDir and runtime configuration for ContentWebApp in render.yaml

farmanahmed888 committed:
Refactor Dockerfile to streamline build process and remove render.yaml configuration

farmanahmed888 commented:
- [x] native login in working but firebase is failing
- [x] requirements.txt replace vonage with vonage<3 to handle login error

farmanahmed888 committed:
Add build-time environment variable configuration for React app in Dockerfile

farmanahmed888 committed:
Update .gitignore files to include .idea directory across multiple projects

farmanahmed888 committed:
Update uvicorn dependency to include standard extras in requirements.txt

farmanahmed888 committed:
Enhance audio playback functions to support external audio URLs and improve error handling

farmanahmed888 committed:
Refactor audio handling to support fallback URLs for local development and improve logging in conference management

farmanahmed888 committed:
Update event URL handling to use BASE_URL instead of NGROK_URL for consistency across actions

Soumabha commented on ConferenceV2/models/system_audio_messages.py:
```diff
@@ -4,34 +4,45 @@
+
+# Fallback URLs for local development (publicly accessible audio files)
+# Using Google Cloud Storage for reliable audio samples
+FALLBACK_BASE_URL = "http://commondatastorage.googleapis.com/codeskulptor-demos/DDR_assets"
```
Lets drive these via the configuration parameters itself.

We can have STORAGE_ENABLED = <azure | googledata>

Soumabha commented on ConferenceV2/models/system_audio_messages.py:
```diff
@@ -4,34 +4,45 @@
 class SystemAudioMessages(str, Enum):
     # Hi, Welcome to SEEDS. Connecting you to the conference call.
-    WELCOME_TEACHER = f"https://{blob_store_name}.blob.core.windows.net/conference/conferenceMessagesWav/english/teacher_welcome_message.wav"
+    WELCOME_TEACHER = f"{base_url}/Kangaroo_MusiQue_-_The_Neverwritten_Role_Playing_Game.mp3" if not azure_storage_enabled else f"{base_url}/teacher_welcome_message.wav"
```
Lets introduce a new table which can keep the mappings between the storage type and the system audio message. Something like

azure | WELCOME_TEACHER | teacher_welcome_message.wav
google | WELCOME_TEACHER | Kangaroo_MusiQue_-.....
.
.
.

Soumabha commented on ConferenceV2/services/singletons/conference_call_manager.py:
```diff
@@ -42,15 +43,18 @@ def create_conference(self, teacher_phone: str, student_phones: List[str]) -> Co
    
-    async def start_conference_call(self, conf_id: str) -> ConferenceCall:
-        conf: ConferenceCall = self.get_conference(conf_id)
+    async def start_conference_call(self, conf_id: str) -> None:
```
It would be good to return the ConferenceCall object itself over here. The reason for that is initially we are trying to fetch an existing conference via the conf_id but if that does not exist, we are creating an absolute new one whose details might be required by the calling function.

Usually 'create' methods of the CRUDs return entire object.

Soumabha commented on ConferenceV2/services/singletons/sas_gen.py:
```diff
@@ -20,31 +25,41 @@ def get_blob_service_client(self, url):
+            
+            # Generate SAS token using the user delegation key
+            sas_token = generate_blob_sas(
+                account_name=blob_service_client.account_name or "",
```
Generate SAS token will always fail for "" account_name, lets put required validation above to ensure that the account_name passed always has requisite value.

Soumabha commented on ConferenceV2/services/singletons/sas_gen.py:
```diff
@@ -20,31 +25,41 @@ def get_blob_service_client(self, url):
+                container_name=container_name,
+                blob_name=blob_path,
+                permission=BlobSasPermissions(read=True),
+                expiry=self.key_expiry_time or datetime.datetime.utcnow() + datetime.timedelta(hours=1),
```
Lets keep the custom expiry configurable via environment so that we can have different values for secure environments like production and relaxed values for staging and test environments.

Soumabha commented on ContentWebApp/Dockerfile:
```diff
@@ -0,0 +1,44 @@
+ARG REACT_APP_FIREBASEMEASUREMENTID
+
+ENV REACT_APP_API_BASE_URL=$REACT_APP_API_BASE_URL
+ENV REACT_APP_FIREBASEAPIKEY=$REACT_APP_FIREBASEAPIKEY
```
Do we need the Firebase values for the ContentWebApp?

Soumabha commented on ContentWebApp/docs/deployment.md:
````diff
@@ -0,0 +1,52 @@
+```
+REACT_APP_API_BASE_URL=<your-backend-api-url>
+REACT_APP_FIREBASEAPIKEY=<your-firebase-api-key>
+REACT_APP_FIREBASEAUTHDOMAIN=<your-firebase-auth-domain>
````
Please validate if we need a Firebase setup for the ContentWebApp. As far as I understand we have removed the mandatory Firebase project setup for running SEEDS.

Soumabha commented on IVRv2/actions/vonage_actions/vonage_action_factory.py:
```diff
@@ -27,7 +27,7 @@ def get_action_implmentation(self, action: Action):
         elif isinstance(action, InputAction):
             return VonageInputAction(type_=action.type,
-                                    eventUrl=os.getenv("NGROK_URL") + action.eventApi,
+                                    eventUrl=os.getenv("BASE_URL") + action.eventApi,
```
The environment file currently suggests a SEEDS_BASE_URL and a BASE_URL. Are these both identical? Maybe we should name these two differently to be more in line with the folder names and prevent confusion.

Soumabha commented on IVRv2/docs/deployment.md:
```diff
@@ -0,0 +1,61 @@
+AZURE_STORAGE_CONNECTION_STRING=<your-azure-storage-connection-string>
+MONGODB_URI=<your-mongodb-atlas-connection-string>
+PYTHON_ENV=staging
+AZURE_BLOB_STORAGE=<your-azure-blob-storage>
```
The environment does not seem to contain BASE_URL or SEEDS_BASE_URL but I believe the codebase uses both of them.

Soumabha commented on teacher-webapp/src/services/apiService.js:
```diff
@@ -60,8 +60,12 @@ export const unmuteParticipant = async (confId, phone_number) => {
 export const playAudio = async (confId) => {
-  const url =  `https://${process.env.REACT_APP_STORAGE_ACCOUNT_NAME}.blob.core.windows.net/output-container/25/1.0.wav`
-  return fetch(`${api_base}/playaudio/${confId}?url=${url}`, {
+  // Use fallback URL if storage account name is not configured
```
The UI should not be making decisions on which audio to be played and that is best left with backend. 

Instead of modifying here to choose the audio url -> which would lead to another configuration for STORAGE_ENABLED = <azure | google>; Lets set a base value for url in the backend itself depending on the storage configuration.

Soumabha commented on teacher-webapp/src/state.js:
```diff
@@ -34,11 +34,13 @@ export class AudioContentState {
 // Sample data for teachers and students
 export const teachers = [
```
Considering that the repository is public, lets move the phone data (PII) to backend database driven invocations with the UI's responsibility to only render the detailed info.

Soumabha commented on websocket-service/docs/deployment.md:
```diff
@@ -0,0 +1,50 @@
+curl -X GET https://your-staging-url.onrender.com/health
```
Seems a bit out of place to have the cURL at the start of README, we can put it in the Testing Endpoints section.

Soumabha commented on websocket-service/src/services/controlService.js:
```diff
@@ -42,12 +42,40 @@ function handleControlMessage(controlMessage) {
+      if (!connection) {
+        console.log(`No connection found for ID ${websocketId}, creating mock connection for testing`);
+        // Create a mock WebSocket connection for testing purposes
+        const mockWs = {
```
No mocks in the actual codebase please. In case we want to be confident about the working, lets write something similar within a test file.

Soumabha commented:
@farmanahmed888 great job on optimising the docker image files for all the codebases. The image sizes now are very manageable and in MBs rather than GBs

farmanahmed888 committed:
Update URLs to point to staging environment for audio and content resources

- Changed pullMenuMainUrl and content_url in multiple files to use the staging blob storage URL.
- Updated audio URLs in quiz_new data structure to reference the staging environment.
- Modified BlobService to connect to the staging blob storage.
- Adjusted API service to use the correct blob storage URL for audio playback.
- Cleaned up unnecessary code in controlService and websocketService related to external URL handling.
- Removed redundant checks and logging for Azure Blob URLs.

Soumabha commented:
@farmanahmed888 there do not seem to be any changes committed post the last review

farmanahmed888 committed:
Add SAS expiry hours configuration and update blob service client expiry time

Soumabha commented on ConferenceV2/models/system_audio_messages.py:
```diff
@@ -4,45 +4,34 @@
-    # Use fallback URLs for local development
-    base_url = FALLBACK_BASE_URL
+if not blob_store_name:
+    raise ValueError("STORAGE_ACCOUNT_NAME environment variable is not set.")
```
Can we also then set "https://{blob_store_name}.blob.core.windows.net/conference/conferenceMessagesWav/english" as the "base_url". 

Currently there is a lot of repetition in the code due to the base_url at each SystemAudioMessages.

Soumabha commented on ConferenceV2/services/singletons/conference_call_manager.py:
```diff
@@ -43,18 +42,15 @@ def create_conference(self, teacher_phone: str, student_phones: List[str]) -> st
         conference_call.set_participant_state(teacher_phone, student_phones)
         self.conferences[conf_id] = conference_call
-        logger_instance.info(f"Conference created and stored. Current conferences: {list(self.conferences.keys())}")
         return conf_id
```
Even though the return type is changed, it seems that we are returning only `conf_id` instead of the `conference_call` object.

Soumabha commented on IVRv2/comprehension_model_classes.py:
```diff
@@ -133,7 +133,7 @@ def get_initial_state(self, initial_quiz_state_id):
 sub_comprehension_evaporation = SubComprehension(
     passage=URLTextEntity(
-        url= 'https://seedsblob.blob.core.windows.net/output-container/comprehension/Evaporation%20-%20intro%20123%20optionsFINAL.mp3',
+        url= 'https://seedsblobstaging.blob.core.windows.net/output-container/comprehension/Evaporation%20-%20intro%20123%20optionsFINAL.mp3',
```
This wont work since we will have a separate blob for production and a separate one for staging. Lets make it into a defined `getBaseUrl()` fuctionality.

farmanahmed888 committed:
Refactor authentication handling: remove Firebase integration, streamline login process, and enhance password validation (#14)

farmanahmed888 committed:
Remove Firebase environment variables from Dockerfile to streamline build process

farmanahmed888 committed:
Return the ConferenceCall instance instead of conf_id in create_conference method

farmanahmed888 committed:
Remove Firebase environment variables from deployment guide to streamline setup

farmanahmed888 committed:
Remove unnecessary curl command from staging deployment guide

farmanahmed888 committed:
Refactor storage account URLs and settings integration

- Updated ivr_constants.py, radio_instantiation.py, url_paths.py, and ivr.py to dynamically construct URLs using the storage account name from settings.
- Added error handling for missing STORAGE_ACCOUNT_NAME in settings.
- Introduced settings.py to manage application configuration using Pydantic.
- Updated BlobService.js to utilize the storage account name from environment variables.
- Added pydantic-settings to requirements.txt for configuration management.

farmanahmed888 committed:
Remove hardcoded DB_CONNECTION from tenant test file

farmanahmed888 committed:
Update SAS expiry time to use configurable hours

farmanahmed888 committed:
Fix SAS token expiry time calculation
